internal/contractgateway/smartcontractgw_test.go

```diff
@@ -1688,6 +1688,41 @@ func TestDeleteSub(t *testing.T) {
 	assert.Equal(204, res.Result().StatusCode)
 }
 
+func TestAddSub(t *testing.T) {
+	assert := assert.New(t)
+
+	mockSubMgr := &mockSubMgr{
+		sub: &events.SubscriptionInfo{
+			Name: "mysub",
+		},
+	}
+	var resBody events.SubscriptionInfo
+	res := testGWPathBody("POST", events.SubPathPrefix, &resBody, mockSubMgr, bytes.NewReader([]byte(`
+    {
+      "name": "mysub"
```

Comment:
Would be nice to see the test pass in all of the possible fields (like `Event` and `Address`) just to validate the expected format for them.

Reply:
Done